src/components/Image/Image.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import browserDimensions from '../../utils/browserDimensions';
+
+const Img = styled.img`
+  width: ${(props) => props.width};
+  height: ${(props) => props.height};
+  position: ${(props) => props.isLoaded ? 'static' : 'initial'};
+  visibility: ${(props) => props.src ? 'initial' : 'hidden'};
+  filter: ${(props) => props.grayscale ? 'grayscale(1)' : 'grayscale(0)'};
+  background-color: ${(props) => props.isLoaded ? 'transparent' : props.theme.color.translucent};
+`;
+
+class Image extends Component {
+  static getCdnUrl(src = '', aspectRatio) {
+    return src ? `${src}?w=${aspectRatio.width * 100}&h=${aspectRatio.height * 100}&fm=pjpg&fit=crop&auto=compress` : src;
+  }
+
+  state = {
+    src: typeof window !== 'undefined' && this.props.lazy ? '' : this.props.src,
+    shouldFetch: (typeof window === 'undefined' && this.props.lazy),
+    isLoaded: (typeof window === 'undefined' && this.props.lazy),
+  };
+
+  componentDidMount() {
+    window.setTimeout(this.handleScroll);
+    window.addEventListener('scroll', this.handleScroll, { passive: true });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { shouldFetch } = this.state;
```

Comment:
`componentWillReceiveProps` will never be called when the state inside the component changes. If there's any change in the parent, the props flows down and triggers `componentWillReceiveProps`.